### ISSUE TITLE
Make the replica setting automatic / more intuitive

### ISSUE THREAD

**Issue by runarmyklebust** [enonic/xp#7565]
The replica-setting today is a bit of a pain; its not easy to remember to set this, and the consequence of not setting the correct number of replicas is high (less performant, less robust on node failure etc).

Research if its feasible to use the automatic replica-setting of ES: `index.auto_expand_replicas`

**Comment by sigdestad** [comment]
Maybe a simpler solution is to change default value to specify default value to 9 replicas? At least then things will work for basic clusters.

**Comment by sigdestad** [comment]
Also, we need to be able to get info about repicas in CLI. Currently you can only set, not see.

**Comment by rymsha** [comment]
> Maybe a simpler solution is to change default value to specify default value to 9 replicas? 

That would lead to unresponsive server, unfortunately. Elasticsearch does not like to be in yellow state.

**Comment by sigdestad** [comment]
What about Runars suggestion for "auto-expanding" replicas?

**Comment by rymsha** [comment]
> What about Runars suggestion for "auto-expanding" replicas?

it is worth trying for sure.
Some caveats:
- it is disabled by default, there must be a reason for that. One reason for sure is immediate increased network traffic when cluster is growing.
- It is not well specified how it plays with number_of_replicas setting we do specify per index. Need to experiment with it (@gbbirkisson your turn!)

**Comment by gbbirkisson** [comment]
```
index.number_of_replicas = 1
index.auto_expand_replicas = 2-all
```

With two initial data nodes, replica count stays at 1. As soon as another data node joins the cluster the replicas scale to 2.
They continue to scale up when other data nodes join. If you remove the data nodes, expected replicas do not go below 2.

```
index.number_of_replicas = 4
index.auto_expand_replicas = 1-3
```

I start a cluster with 2 nodes that has the default replica count of 1. If I try to set replicas to 4, it will be immediately overwritten by `index.auto_expand_replicas` and it sets them to 1 again.

**Comment by rymsha** [comment]
So, this is a feature that was always present, just not documented

**Comment by sigdestad** [comment]
Interesting, but does this take into account data nodes vs non-data nodes?

**Comment by gbbirkisson** [comment]
Yes, ignores non-data nodes. Replica scaling only happens when number of data nodes change. I suspect this `all` parameter is really `all-data-nodes`.

**Comment by sigdestad** [comment]
It would simplify life for many customers if this could be enabled. I guess even for the operator?

**Comment by gbbirkisson** [comment]
Yup. I have already added this to the operator for testing.

**Comment by rymsha** [comment]
> Interesting, but does this take into account data nodes vs non-data nod

It has some known limitations described in ED documentation

> Note that the auto-expanded number of replicas only takes allocation filtering rules into account, but ignores any other allocation rules such as shard allocation awareness and total shards per node, and this can lead to the cluster health becoming YELLOW if the applicable rules prevent all the replicas from being allocated.

But we don't use these features.

**Comment by rymsha** [comment]
@gbbirkisson hopefully you can update docs, so we can close this issue.

**Comment by gbbirkisson** [comment]
Busy these days, you could always do it your self @rymsha.
